vpnglean/CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+add_library(vpnglean STATIC IMPORTED GLOBAL)
+
+set_target_properties(vpnglean PROPERTIES FOLDER "Libs")
+
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+    set(OPENSSL_USE_STATIC_LIBS TRUE)
+    find_package(OpenSSL REQUIRED)
+    cmake_path(GET OPENSSL_SSL_LIBRARY PARENT_PATH OPENSSL_LIB_DIR)
+endif()
+
+set(CARGO_BUILD_ENV
+    BUILD_ID=${BUILD_ID}
+    APP_VERSION=${APP_VERSION}
+    CARGO_TARGET_DIR=${CMAKE_CURRENT_BINARY_DIR}
+    OPENSSL_STATIC=yes
+    OPENSSL_LIB_DIR=${OPENSSL_LIB_DIR}
+    OPENSSL_INCLUDE_DIR=${OPENSSL_INCLUDE_DIR}
+)
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(CARGO_CMD ${CARGO_BUILD_ENV} cargo build)
+    set(TARGET_DIR "debug")
+else ()
+    set(CARGO_CMD ${CARGO_BUILD_ENV} cargo build --release)
+    set(TARGET_DIR "release")
+endif ()
+
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
```

Comment:
This can be done platform agnostically with:
`set(LIBNAME ${CMAKE_STATIC_LIBRARY_PREFIX}vpnglean${CMAKE_STATIC_LIBRARY_SUFFIX})`